gwnr/waveform/hybridize.py

```diff
@@ -0,0 +1,177 @@
+        ''' i1 is an integer, which is the numerical value of the index value where frq_desired is located, 
+        it is extracted from the tuple where_i1'''
+        return i1
+    
+    
+    
+    t1_index_insp = find_i1(frq_insp, frq_attach - frq_width/2)
+    t2_index_insp = find_i1(frq_insp, frq_attach + frq_width/2)
+
+
+    t1_index_mr_tent = find_i1(frq_mr, frq_attach - frq_width/2)
+    # t2_index_mr_tent = find_i1(frq_mr, frq_attach + frq_width/2) 
+
+    # another way to define t2_index_mr_tent is through number of points in the inspiral window    
+    t2_index_mr_tent = t1_index_mr_tent + (t2_index_insp-t1_index_insp)
+
+    no_cycles = (phase_insp[t1_index_insp] - phase_insp[t2_index_insp])/(2*np.pi)
+
+
+    ''' After alignment, need to find corresponding indices for merger_ringdown as well '''
+    
+    
+    ''' Alignment: '''
+    
+    # Defining the Comb
+    
+    def get_comb(t1_index_insp, t2_index_insp, no_sp):
+
+        d = int((t2_index_insp-t1_index_insp)/no_sp)
+
+        count1 = np.empty(no_sp-1)
+        for kk in range(no_sp-1):
+            count1[kk] = kk*d
+
+        samples = np.append(count1, t2_index_insp-t1_index_insp-1).astype(int)
+        return samples
+
+    sample_indices_insp = get_comb(t1_index_insp, t2_index_insp, no_sp = 4)
+    sample_indices_mr = get_comb(t1_index_mr_tent, t2_index_mr_tent, no_sp = 4)
+
+
+
+    ''' Enter alignment code here '''
+
+    ''' Need interpolated frequency evaluated on the time axis for timeshifts '''
+
+    interp_phase_insp = UnivariateSpline(timeaxis_insp[t1_index_insp:t2_index_insp], phase_insp[t1_index_insp:t2_index_insp], k = 4)
+    interp_phase_mr = UnivariateSpline(timeaxis_mr[t1_index_mr_tent:t2_index_mr_tent], phase_mr[t1_index_mr_tent:t2_index_mr_tent], k = 4)
+
+    interp_frq_insp = interp_phase_insp.derivative()(timeaxis_insp[t1_index_insp:t2_index_insp])/(2*np.pi)
+    interp_frq_mr = interp_phase_mr.derivative()(timeaxis_mr[t1_index_mr_tent:t2_index_mr_tent])/(2*np.pi)
+
+
+
+
+
+
+
+
+
+    ''' alignment using corrective phase addition '''
+
+    def mismatch_discrete(w1, w2):
+        w1_d = w1[sample_indices_insp]
+        w2_d = w2[sample_indices_mr] # can't give the same comb to w2
+        w1sq = np.square(np.abs(w1_d))
+        w2sq = np.square(np.abs(w2_d))
+        diff = np.abs(w1_d - w2_d)
+        diffsq = np.square(diff)
+        mm = 0.5*(np.sum(diffsq)/np.sum(w1sq))
+        return mm
+    '''
+    def optfn_ph(phaseshift_correction):
+    phase_corr_mr = merger_ringdown*np.exp(1j*phaseshift_correction)
+    m = mismatch_discrete(inspiral[t1_index_insp:t2_index_insp], phase_corr_mr[t1_index_mr_tent:t2_index_mr_tent])
+    return m
+    '''
+
+    ''' Performing attachment using the blending function '''
+
+    # blending fn is an array
+    blfn_var = np.arange(t1_index_insp, t2_index_insp)
+    tau = np.square((np.sin((np.pi/2)*(blfn_var-t1_index_insp)/(t2_index_insp-t1_index_insp))))
+
+
+    def blend1(x1, x2):
```

Comment:
- move the calculation of `tau` inside this function. Take the start/end indices of BOTH series that are being blended as inputs.
- rename to `blend_series`
- move outside the main function